### ISSUE TITLE
bug/check for magic library availability doesn't appear to be correct

### ISSUE THREAD

**Issue by aaronsteers** [Unstructured-IO/unstructured#3781]
**Describe the bug**

Currently, it appears that libmagic is detected with this line:

```py
LIBMAGIC_AVAILABLE = bool(importlib.util.find_spec("magic"))
```

This finds whether `magic` is installed, but that's not the correct check, I think. Since `python-magic` in included in the `unstructured` dependencies, the library will be installed at the same time as Unstructured. However, if the `magiclib` c library is not available, those implementations will fail.

**To Reproduce**

On my Mac, I'm running this code:

```py
        try:
            file_type = detect_filetype(
                file_path=remote_file.uri,
            )
        except Exception:
            # Path doesn't exist locally. Fall back to using an extension-based detection.
            file_type: FileType | None = FileType.from_extension(
                remote_file.uri.split(".")[-1],
            )
```

**Expected behavior**

Expected to fail over to other means of detection if the magiclib c library is not available.

**Screenshots**

**Environment Info**
Please run `python scripts/collect_env.py` and paste the output here. 
This will help us understand more about the environment in which the bug occurred.

**Additional context**

Log is here:

```
 File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/unstructured/file_utils/filetype.py", line 131, in file_type
    return cls(ctx)._file_type
  File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/unstructured/file_utils/filetype.py", line 141, in _file_type
    if file_type := self._file_type_from_guessed_mime_type:
  File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/unstructured/file_utils/filetype.py", line 181, in _file_type_from_guessed_mime_type
    mime_type = self._ctx.mime_type
  File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/unstructured/utils.py", line 154, in __get__
    value = self._fget(obj)
  File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/unstructured/file_utils/filetype.py", line 362, in mime_type
    import magic
  File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/magic/__init__.py", line 209, in <module>
    libmagic = loader.load_lib()
  File "/Users/ajsteers/Source/my-project/.venv/lib/python3.10/site-packages/magic/loader.py", line 49, in load_lib
    raise ImportError('failed to find libmagic.  Check your installation')
ImportError: failed to find libmagic.  Check your installation
```

**Comment by scanny** [comment]
@aaronsteers Hmm, interesting. That makes sense to me.

Out of curiosity, what platform are you running on where you don't have or don't want `libmagic` installed?

It turns out that our use of `libmagic` has become more marginal because we need more reliable (if perhaps micro-seconds less efficient) detection of binary types and so have largely rolled our own. So it could be the right fix is just to remove this dependency.

**Comment by aaronsteers** [comment]
Hi, @scanny ! Thanks for the quick reply!

> Out of curiosity, what platform are you running on where you don't have or don't want libmagic installed?

I'm doing development on a Mac M2, and planning to deploy to a Ubuntu image. But dragging my feet on the obvious `brew install magiclib` because (1) I'm not 100% sure the ubuntu image has this library and (2) it means others who try to run the same as me would need to pre-install with brew before they could contribute.

It's very possible that my Mac laptop _does_ have this library but perhaps it isn't found? Curious if other Mac users are able to repo or if the library missing is more of an "on my machine" problem... 🤔

**Comment by metadaddy** [comment]
I just arrived here via Google with the exact same issue as @aaronsteers. Working on a Mac M1. `libmagic` is definitely not present on my machine:

```console
% mdfind -name libmagic
2024-11-21 08:53:34.665 mdfind[78897:6068546] [UserQueryParser] Loading keywords and predicates for locale "en_US"
2024-11-21 08:53:34.666 mdfind[78897:6068546] [UserQueryParser] Loading keywords and predicates for locale "en"
/Users/ppatterson/src/php-src/ext/fileinfo/libmagic
/Users/ppatterson/src/php-src/ext/fileinfo/libmagic.patch
/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/libmagic.rb
```

I'm gonna do `brew install libmagic` for now, and add it to the prereqs for the app I'm creating 😕

I think either:
- The `magic` dependency should be removed, or
- The `import magic` line should be in a `try` block, since there is a fallback.

In fact, I'll see if I can spin up a PR with the latter...

**Comment by scanny** [comment]
This should be resolved by #3828 which should merge next week and be in next release.z
